### PR TITLE
Add pps2021 support

### LINKED ISSUE #27: Add a general level1c runner to the repo
#### Code Sample, a minimal, complete, and verifiable piece of code

```python
# Your code here

```
#### Problem description

Add a general (supporting any imager that the https://github.com/foua-pps/level1c4pps supports) level1c runner to this repo. 

There is a seviri-level1c runner already, but add this code here, as it is anyhow needed for future PPS versions (version 2021 and onwards).

#### Versions of Python, package at hand and relevant dependencies

Thank you for reporting an issue !

### PR DESCRIPTION
**Pull request by adybbroe**
Adding support for running PPSv2021

 - [x] Closes #27  
 - [x] Tests added 
 - [x] Tests passed: Passes ``pytest nwcsafpps_runner`` 
 - [ ] Passes ``flake8 nwcsafpps_runner`` 
 - [ ] Fully documented

### CODE REVIEW

**Comment by TAlonglong** [comment]
Nice work @adybbroe. Please let me know when its ready for review

File: bin/pps2018_runner.py [hunk 1]
```diff
@@ -114,23 +116,22 @@ def pps_worker(scene, publish_q, input_msg, options):
...
+        py_exec = options.get('python')
+        pps_run_all = options.get('run_all_script')
+        pps_script = pps_run_all.get('name')
+        pps_run_all_flags = pps_run_all.get('flags')
+        if not pps_run_all_flags:
+            pps_run_all_flags = []
+
+        use_l1c = (options.get('pps_version') == 'v2021')
+        cmd_str = create_pps_call_command(py_exec, pps_script, scene, use_l1c=use_l1c)
         run_cpp = options.get('run_pps_cpp', None)
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply bin/pps2018_runner.py:129]: Yes, you are right, it is not used now. I have removed it.

File: bin/pps2018_runner.py [hunk 2]
```diff
@@ -114,23 +116,22 @@ def pps_worker(scene, publish_q, input_msg, options):
...
-        if not run_cpp:
-            cmd_str = cmd_str + ' --no_cpp'
+        # if not run_cpp:
+        #    cmd_str = cmd_str + ' --no_cpp'
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply bin/pps2018_runner.py:131]: Done

File: bin/pps2018_runner.py [hunk 3]
```diff
@@ -366,8 +331,13 @@ def pps(options):
                            sdr_granule_processing=options.get('sdr_processing') == 'granules')
         if status:
             sceneid = get_sceneid(platform_name, orbit_number, starttime)
-            scene['file4pps'] = get_pps_inputfile(platform_name, files4pps[sceneid])
+            LOG.debug(files4pps[sceneid])
+            if not use_l1c:
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply bin/pps2018_runner.py:335]: Thx, done.

File: bin/pps2018_runner.py [hunk 4]
```diff
@@ -403,10 +373,7 @@ def pps(options):
 if __name__ == "__main__":
 
     from logging import handlers
-    LOG.debug("Path to pps2018_runner config file = " + CONFIG_PATH)
-    LOG.debug("Pps2018_runner config file = " + CONFIG_FILE)
-    configfile = os.path.join(CONFIG_PATH, CONFIG_FILE)
-    OPTIONS = get_config(configfile)
+    OPTIONS = get_config(os.path.join(CONFIG_PATH, CONFIG_FILE))
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply bin/pps2018_runner.py:376]: You are right. Would be better/cleaner to provide the config-file-path on the commanline. I might have a go at that...

File: bin/pps2018_runner.py [hunk 5]
```diff
@@ -434,5 +401,7 @@ def pps(options):
     logging.getLogger('posttroll').setLevel(logging.INFO)
 
     LOG = logging.getLogger('pps_runner')
+    LOG.debug("Path to pps2018_runner config file = " + CONFIG_PATH)
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply bin/pps2018_runner.py:404]: Fixed

File: nwcsafpps_runner/config.py [hunk 6]
```diff
@@ -105,18 +105,20 @@ def get_config_init_cfg(configfile, service=MODE):
 
 def get_config_yaml(configfile, service=MODE, procenv=''):
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply nwcsafpps_runner/config.py:106]: Agree! I have removed it now. At SMHI we can deal with this now. We have sym-links on the servers pointing to the path where we have the model data. The model data have different paths depending on dev, test and production, but the links on each server environment have the same name.

File: nwcsafpps_runner/config.py [hunk 7]
```diff
@@ -105,18 +105,20 @@ def get_config_init_cfg(configfile, service=MODE):
...
-        from yaml import Loader as UnsafeLoader
+    # import yaml
+    # try:
+    #     from yaml import UnsafeLoader
+    # except ImportError:
+    #     from yaml import Loader as UnsafeLoader
 
-    with open(configfile, 'r') as fp_:
-        config = yaml.load(fp_, Loader=UnsafeLoader)
+    # with open(configfile, 'r') as fp_:
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply nwcsafpps_runner/config.py:114]: Yes, done. I have now also removed all code related to handling ini files

**Comment by adybbroe** [comment]
I have followed most (if not all) of your suggestions now. I need to check that it still works locally (system testing) with v2018. Also, would be good to verify again that it works for v2021 as well.
Once that is verified positively I agree we should rename the runner to `pps_runner` (replacing the old one).

File: nwcsafpps_runner/config.py [hunk 8]
```diff
@@ -105,18 +105,20 @@ def get_config_init_cfg(configfile, service=MODE):
 
 def get_config_yaml(configfile, service=MODE, procenv=''):
     """Get the configuration from file."""
-    import yaml
-    try:
-        from yaml import UnsafeLoader
-    except ImportError:
-        from yaml import Loader as UnsafeLoader
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply nwcsafpps_runner/config.py:112]: Done

**Comment by adybbroe** [comment]
This version is now tested with v2018 and AVHRR/MODIS. There is something now here that doesn't work with VIIRS, but not sure what it is yet. Haven't had the time yet. I suggest we merge this one, and make smaller PRs for what might remain to make it work for all instruments and both PPS versions. Including the renaming of the pps2018_runner script.

**Pull request merged**